### PR TITLE
Add dark mode toggle switch

### PR DESCRIPTION
**Pull request by PikaMug**
Shown in #2826 not urgent / low priority

Debating whether to remove TOGGLE_DARK_MODE_TEXT or show it on hover

### CODE REVIEW

File: custom/templates/DefaultRevamp/footer.tpl [hunk 1]
```diff
@@ -12,7 +12,19 @@
                     {if $PAGE_LOAD_TIME}
                     <span class="item" id="page_load"></span>
                     {/if}
-                    <a class="item" href="javascript:" onclick="toggleDarkLightMode()">{$TOGGLE_DARK_MODE_TEXT}</a>
+                    <input type="checkbox" class="darkmode-toggle" id="darkmode-toggle" onclick="toggleDarkLightMode()">
+                    <label for="darkmode-toggle" class="darkmode-toggle-label">
+                        <i class="fas fa-moon"></i>
+                        <i class="fas fa-sun"></i>
+                        <div class="darkmode-ball"></div>
+                    </label>
```
Comment (PadowYT2) [inline custom/templates/DefaultRevamp/footer.tpl:20]:
```html
                    <label for="darkmode-toggle" class="darkmode-toggle-label">
                        <i class="fas fa-moon"></i>
                        <i class="fas fa-sun"></i>
                        <div class="darkmode-ball"></div>
                        {$TOGGLE_DARK_MODE_TEXT}
                    </label>```

File: custom/templates/DefaultRevamp/footer.tpl [hunk 2]
```diff
@@ -12,7 +12,19 @@
...
+                    <script type="text/javascript">
+                        if (document.body.classList.contains('dark')) {
+                            document.getElementById("darkmode-toggle").checked = true;
+                        } else {
+                            document.getElementById("darkmode-toggle").checked = false;
+                        }
+                    </script>
```
Comment (PadowYT2) [inline custom/templates/DefaultRevamp/footer.tpl:27]: Should be at the bottom

**Review by PadowYT2 — CHANGES_REQUESTED** [review]

File: custom/templates/DefaultRevamp/footer.tpl (hunk 2 above)
Comment (PikaMug) [reply custom/templates/DefaultRevamp/footer.tpl:27]: I'd like to hear if other contributors agree as this script isn't meant to be global, so to speak.

**Comment by Aberdeener** [comment]
Delaying testing/merging of this until 2.1.0 given the breaking change this would be to templates 🙏🏼

File: custom/templates/DefaultRevamp/footer.tpl (hunk 2 above)
Comment (samerton) [reply custom/templates/DefaultRevamp/footer.tpl:27]: I'm happy for this to be inline like this 👍

**Review by samerton — APPROVED** [review]

**Comment by samerton** [comment]
Thanks for this!

**Pull request merged**